### PR TITLE
fix(NX-3544): add automated messages for inquiries

### PR DESCRIPTION
**Pull request by tam-kis**
This PR resolves [NX-3544](https://artsyproduct.atlassian.net/browse/NX-3544)

### Description

Add automated messages for inquiries.
Implemented in Force: [NX-3543](https://artsyproduct.atlassian.net/browse/NX-3543)

https://github.com/artsy/eigen/assets/17580625/98b9d68d-8f27-426e-adcd-3fe29516deae

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-  add automated messages for inquiries - tam-kis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[NX-3544]: https://artsyproduct.atlassian.net/browse/NX-3544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NX-3543]: https://artsyproduct.atlassian.net/browse/NX-3543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tests.tsx [hunk 1]
```diff
@@ -151,8 +151,6 @@ describe("<InquiryModal />", () => {
     await press(wrapper.root, { text: "Cancel" })
 
     expect(wrapper.root.findByType(InquiryModalFragmentContainer).props.modalIsVisible).toBeFalsy()
-    expect(checkBox.props.checked).toBeFalsy()
-    expect(input.props.value).toBeFalsy()
```
Comment (tam-kis) [inline src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tests.tsx:155]: The modal is unmounted so we are unable to find the checkbox in input. This file has a lot of deprecations so it need to be updated - let me know if we should have this as a separate ticket.

Comment (gkartalis) [reply src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tests.tsx:155]: happy to open another PR to update the tests 🤘

File: src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx [hunk 2]
```diff
@@ -154,10 +154,62 @@ export const InquiryModal: React.FC<InquiryModalProps> = ({ artwork, ...props })
   const [loading, setLoading] = useState(false)
   const selectShippingLocation = (locationDetails: LocationWithDetails) =>
     dispatch({ type: "selectShippingLocation", payload: locationDetails })
-  const setMessage = (message: string) => dispatch({ type: "setMessage", payload: message })
+  const setMessage = useCallback(
+    (message: string) => dispatch({ type: "setMessage", payload: message }),
+    [dispatch]
+  )
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:160]: 👏

File: src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx [hunk 3]
```diff
@@ -154,10 +154,62 @@ export const InquiryModal: React.FC<InquiryModalProps> = ({ artwork, ...props })
...
+    "Hi, I'm very interested in this artwork and would appreciate more information about it.",
+    "Hi there, I'm interested in this artwork and would like to know more about it. Can you provide further information?",
+    "Hello, I'm a collector and would love to learn more about this artwork. Could you please share additional details?",
+    "Hi, as someone who collects art, I was intrigued by this piece. Can you provide more information about it?",
+    "Hi, I'm a collector and would like to know more about this artwork. Can you please provide additional information?",
+  ]
+
+  const getAutomatedMessages = () => {
+    return AUTOMATED_MESSAGES[Math.floor(Math.random() * AUTOMATED_MESSAGES.length)]
+  }
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:208]: 🔥

File: src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx [hunk 4]
```diff
@@ -154,10 +154,62 @@ export const InquiryModal: React.FC<InquiryModalProps> = ({ artwork, ...props })
...
+    "Hi there. I came across this artwork today. Could you please provide me with more information about it?",
+    "Hello, I'm interested in this artwork and would love to know more about it.",
+    "Hi, I collect art and was intrigued by this artwork. Can you provide additional information?",
+    "Hello, I'm a collector and would like to inquire about this artwork. Could you please share more details?",
+    "Hi, I'm very interested in this artwork and would appreciate more information about it.",
+    "Hi there, I'm interested in this artwork and would like to know more about it. Can you provide further information?",
+    "Hello, I'm a collector and would love to learn more about this artwork. Could you please share additional details?",
+    "Hi, as someone who collects art, I was intrigued by this piece. Can you provide more information about it?",
+    "Hi, I'm a collector and would like to know more about this artwork. Can you please provide additional information?",
+  ]
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:204]: Amazing! Question though: What about moving this const outside of the component?

A good place would be `Artwork/Components/CommercialButtons/constants.ts` and import it from there to use it here.

**Review by gkartalis — DISMISSED** [review]
Minor suggestion about moving the messages to another file but looks good!

File: src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx (hunk 4 above)
Comment (tam-kis) [reply src/app/Scenes/Artwork/Components/CommercialButtons/InquiryModal.tsx:204]: Yes, that's a great idea! Didn't know about these files in Eigen - it's much better to keep this entire array outside the component file and then import it.

**Review by gkartalis — APPROVED** [review]

**Review by araujobarret — COMMENTED** [review]
:ceiling_cat: looks good!

**Pull request merged**